### PR TITLE
new script emojc2ascii: converts emojis into ascii

### PR DESCRIPTION
**Pull request by eyJhb**
## Script info
Added upon request from https://github.com/eyJhb/weechat-emoji2ascii/issues/1

- Script name: emoji2ascii.py
- Version: 0.1

- Requirements: emoji

- Min WeeChat version: unsure

- Script tags: python, irc, emoji, ascii, py3

## Description

replaces incoming emojis with ascii text, and can replace outgoing, messages with emojis, if the same syntax is followed.

## Checklist (new script)

- [x] Single commit, single file added
- [x] Commit message: `New script name.py: short description…`
- [?] No similar script already exists (emoji has lots of similar names)
- [x] Name: max 32 chars, only lower case letters, digits and underscores
- [x] Unique name, does not already exist in repository
- [x] No shebang on the first line
- [x] Comment in script with name/pseudo, e-mail and license
- [x] Only English in code/comments
- [?] Pure WeeChat API used, no extra API
- [?] Function `hook_process` is used for any blocking call
- [x] For Python script: works with Python 3 (Python 2 support is optional)

## Checklist (script update)

- [ ] Author has been contacted
- [ ] Single commit, single file added
- [ ] Commit message format: `script_name.py X.Y: …`
- [ ] Script version and Changelog have been updated
- [ ] For Python script: works with Python 3 (Python 2 support is optional)

## Checklist (script deletion)

- [ ] Author has been contacted
- [ ] Single commit, single file deleted
- [ ] Commit message format: `Remove script name.py` with reasons in description

### CODE REVIEW

File: python/emoji2ascii.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        w.hook_modifier("irc_in_away", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_cnotice", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_cprivmsg", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_kick", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_knock", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_notice", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_part", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_privmsg", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_quit", "convert_emoji_to_aliases", "")
+        w.hook_modifier("irc_in_wallops", "convert_emoji_to_aliases", "")
```
Comment (flashcode) [inline python/emoji2ascii.py:74]: As mentioned in the scripting guide here: https://weechat.org/files/doc/devel/weechat_scripting.en.html#python_strings, you should replace all these "irc_in_xxx" modifiers by "irc_in2_xxx" which will always receive valid UTF-8 data.
Could you please change this?

**Review by flashcode — CHANGES_REQUESTED** [review]

**Comment by eyJhb** [comment]
I will get right on it. Give me a day, two tops. :)

Thanks for pointing it out!

On 27 May 2021 21:39:23 CEST, "Sébastien Helleu" ***@***.***> wrote:
***@***.*** requested changes on this pull request.
>
>
>
>> +    return emoji.emojize(string)
>+
>+if __name__ == "__main__" and import_ok:
>+    if w.register(SCRIPT_NAME, SCRIPT_AUTHOR, SCRIPT_VERSION,
>SCRIPT_LICENSE,
>+                  SCRIPT_DESC, "", "utf-8"):
>+
>+        w.hook_modifier("irc_in_away", "convert_emoji_to_aliases", "")
>+        w.hook_modifier("irc_in_cnotice", "convert_emoji_to_aliases",
>"")
>+        w.hook_modifier("irc_in_cprivmsg", "convert_emoji_to_aliases",
>"")
>+        w.hook_modifier("irc_in_kick", "convert_emoji_to_aliases", "")
>+        w.hook_modifier("irc_in_knock", "convert_emoji_to_aliases",
>"")
>+        w.hook_modifier("irc_in_notice", "convert_emoji_to_aliases",
>"")
>+        w.hook_modifier("irc_in_part", "convert_emoji_to_aliases", "")
>+        w.hook_modifier("irc_in_privmsg", "convert_emoji_to_aliases",
>"")
>+        w.hook_modifier("irc_in_quit", "convert_emoji_to_aliases", "")
>+        w.hook_modifier("irc_in_wallops", "convert_emoji_to_aliases",
>"")
>
>As mentioned in the scripting guide here:
>https://weechat.org/files/doc/devel/weechat_scripting.en.html#python_strings,
>you should replace all these "irc_in_xxx" modifiers by "irc_in2_xxx"
>which will always receive valid UTF-8 data.
>Could you please change this?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/weechat/scripts/pull/462#pullrequestreview-670570158

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: python/emoji2ascii.py (hunk 1 above)
Comment (eyJhb) [reply python/emoji2ascii.py]: Should be resolved in the latest force push :)

**Review by flashcode — APPROVED** [review]

**Comment by flashcode** [comment]
Thanks, script added in the repository.

**Pull request closed without merging**